qemu/tests/boot_from_remote.py

```diff
@@ -0,0 +1,161 @@
+            finally:
+                vm.destroy()
+
+    def boot_with_local_image():
+        """
+        Boot up a guest with a remote storage system image
+            as well as a local filesystem image
+        VM can start up without any error
+        """
+        try:
+            session = None
+            vm = env.get_vm(params["main_vm"])
+            vm.verify_alive()
+            tm = float(params.get("login_timeout", 240))
+            session = vm.wait_for_login(timeout=tm)
+
+            _write_disk(session)
+        finally:
+            if session:
+                session.close()
+            vm.destroy()
+
+    def boot_with_remote_images():
+        """
+        Boot up a guest with only one remote image,
+            record memory consumption(vsz, rss)
+        Boot up a guest with 4 remote images,
+            record memory consumption(vsz, rss)
+        The memory increased should not be greater than 'memory_diff'
+        """
+        try:
+            vm = env.get_vm(params["main_vm"])
+            vm.verify_alive()
+
+            # get vsz, rss when booting with one remote image
+            single_img_memory = _get_memory(vm.get_pid())
+            if not single_img_memory:
+                raise exceptions.TestError("Failed to get memory when "
+                                           "booting with one remote image.")
+            logging.debug("memory consumption(only one remote image): %s"
+                          % single_img_memory)
+
+            vm.destroy()
+
+            for img in params['images'].split()[1:]:
+                params['boot_drive_%s' % img] = 'yes'
+            env_process.preprocess_vm(test, params, env, params["main_vm"])
+            vm = env.get_vm(params["main_vm"])
+            vm.verify_alive()
+
+            # get vsz, rss when booting with 4 remote image
+            multi_img_memory = _get_memory(vm.get_pid())
+            if not multi_img_memory:
+                raise exceptions.TestError("Failed to get memory when booting"
+                                           " with several remote images.")
+            logging.debug("memory consumption(total 4 remote images): %s"
+                          % multi_img_memory)
+
+            diff = int(float(utils_numeric.normalize_data_size(
+                params['memory_diff'], order_magnitude="K")))
+            mem_diffs = [i-j for i, j in zip(multi_img_memory,
+                                             single_img_memory)]
+            if mem_diffs[0] > diff:
+                raise exceptions.TestFail(
+                    "vsz increased '%s', which was more than '%s'"
+                    % (mem_diffs[0], diff))
+            if mem_diffs[1] > diff:
+                raise exceptions.TestFail(
+                    "rss increased '%s', which was more than '%s'"
+                    % (mem_diffs[1], diff))
+        finally:
+            vm.destroy()
+
+    try:
+        tc = params["scenario"]
+        eval("%s()" % tc)
```

Comment:
`eval` is evil , better to use `locals()[tc]()`

Reply:
done